CodeHub.Service/Services/FollowService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using AutoMapper;
+using CodeHub.DataAccess.Repositories;
+using CodeHub.Domain.Entities;
+using CodeHub.Model.Follow;
+using CodeHub.Service.Exceptions;
+using CodeHub.Service.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace CodeHub.Service.Services;
+
+public class FollowService : IFollowService
+{
+    private readonly IMapper mapper;
+    private readonly IGenericRepository<Follow> repository;
+    private readonly IUserService userService;
+
+    public FollowService(IMapper mapper, IGenericRepository<Follow> repository, IUserService userService)
+    {
+        this.mapper = mapper;
+        this.repository = repository;
+        this.userService = userService;
+    }
+
+    public async Task<FollowViewModel> CreateAsync(FollowCreateModel follow)
+    {
+        if (follow.FollowerId == follow.FollowingId)
+            throw new CustomException(400, "You can't follow yourself");
+
+        var existFollower = await repository.SelectByIdAsync(follow.FollowerId)
```

Comment:
Did you need to use userService instead of repository which is of follow type?

Reply:
you are right